cleanrl/sac_atari.py

```diff
@@ -0,0 +1,359 @@
+
+    # Algorithm specific arguments
+    parser.add_argument("--env-id", type=str, default="Seaquest-v4",
+        help="the id of the environment")
+    parser.add_argument("--total-timesteps", type=int, default=120000,
+        help="total timesteps of the experiments")
+    parser.add_argument("--buffer-size", type=int, default=int(1.5e5),
+        help="the replay memory buffer size") # smaller than in original paper but evaluation is done only for 100k steps anyway
+    parser.add_argument("--gamma", type=float, default=0.99,
+        help="the discount factor gamma")
+    parser.add_argument("--tau", type=float, default=1.0,
+        help="target smoothing coefficient (default: 1)") # Default is 1 to perform replacement update
+    parser.add_argument("--batch-size", type=int, default=64,
+        help="the batch size of sample from the reply memory")
+    parser.add_argument("--learning-starts", type=int, default=2e4,
+        help="timestep to start learning")
+    parser.add_argument("--policy-lr", type=float, default=3e-4,
+        help="the learning rate of the policy network optimizer")
+    parser.add_argument("--q-lr", type=float, default=3e-4,
+        help="the learning rate of the Q network network optimizer")
+    parser.add_argument("--policy-frequency", type=int, default=4,
+        help="the frequency of training policy (delayed)")
+    parser.add_argument("--target-network-frequency", type=int, default=8000, # Denis Yarats' implementation delays this by 2.
+        help="the frequency of updates for the target networks")
+    parser.add_argument("--alpha", type=float, default=0.2,
+        help="Entropy regularization coefficient.")
+    parser.add_argument("--autotune", type=lambda x:bool(strtobool(x)), default=True, nargs="?", const=True,
+        help="automatic tuning of the entropy coefficient")
+    args = parser.parse_args()
+    # fmt: on
+    return args
+
+
+def make_env(env_id, seed, idx, capture_video, run_name):
+    def thunk():
+        env = gym.make(env_id)
+        env = gym.wrappers.RecordEpisodeStatistics(env)
+        if capture_video:
+            if idx == 0:
+                env = gym.wrappers.RecordVideo(env, f"videos/{run_name}")
+        env = NoopResetEnv(env, noop_max=30)
+        env = MaxAndSkipEnv(env, skip=4)
+        env = EpisodicLifeEnv(env)
+        if "FIRE" in env.unwrapped.get_action_meanings():
+            env = FireResetEnv(env)
+        env = ClipRewardEnv(env)
+        env = gym.wrappers.ResizeObservation(env, (84, 84))
+        env = gym.wrappers.GrayScaleObservation(env)
+        env = gym.wrappers.FrameStack(env, 4)
+        env.seed(seed)
+        env.action_space.seed(seed)
+        env.observation_space.seed(seed)
+        return env
+
+    return thunk
+
+
+# He initialization
+def network_init(net, bias_const=0.0):
+    for param in net.parameters():
+        if type(param) == nn.Linear or type(param) == nn.Conv2d:
```

Comment:
I thought I would use this opportunity to learn a bit more, so let me know what you think:

While `type()` does the job, but from an OOP perspective, I feel like using `isinstance(param, nn.Linear)`
would be more adequate.

"type" usually refers to the "data type", life float, int, etc...
But `isinstance()` is expected when working with classes, such as `nn.Linear`.
Although in Python, they essentially work in the same way...

Reply:
From my knowledge I would agree with you that `isinstance()` is more Pythonic. As far as I know it also properly handles inheritance.
I started from the original code which was using `type()` and hadn't changed it. I'll clean this up.